src/ziggurat/messaging/rabbitmq/cluster/producer.clj

```diff
@@ -8,23 +8,33 @@
             [clojure.string :as str])
   (:import (org.apache.kafka.common.header.internals RecordHeader)))
 
+(defn get-default-ha-policy [hosts-vec cluster-config]
+  (let [host-count   (count hosts-vec)
+        ha-mode      (get cluster-config :ha-mode "all")
+        ha-params    (get cluster-config :ha-params host-count)
+        ha-sync-mode (get cluster-config :ha-sync-mode "automatic")]
+    (if (= "all" ha-mode)
+      {:ha-mode ha-mode :ha-sync-mode ha-sync-mode}
+      {:ha-mode ha-mode :ha-sync-mode ha-sync-mode :ha-params ha-params})))
+
 (defn set-ha-policy [queue-name exchange-name cluster-config]
-  (let [hosts (atom (str/split (:hosts cluster-config) #","))]
-    (with-retry {:count (count @hosts)
-                 :wait 50
+  (let [hosts-vec (str/split (:hosts cluster-config) #",")
+        hosts     (atom hosts-vec)]
```

Comment:
I dont think you need the atom here

Reply:
can you suggest another way to mutate the hosts?

Reply:
you dont need to mutate the hosts, you can just remove the atom

Reply:
since it runs inside a `with-retry`, we need to remove one entry everytime we retry.

Reply:
hmmm can this be done inside the *with-retry recursion? passing the tail of the vector until you reach an empty vector...